src/main/cli/settings.c

```diff
@@ -1315,7 +1315,7 @@ const clivalue_t valueTable[] = {
     { "osd_alt_alarm",              VAR_UINT16 | MASTER_VALUE, .config.minmaxUnsigned = { 0, 10000 }, PG_OSD_CONFIG, offsetof(osdConfig_t, alt_alarm) },
     { "osd_distance_alarm",         VAR_UINT16 | MASTER_VALUE, .config.minmaxUnsigned = { 0, UINT16_MAX }, PG_OSD_CONFIG, offsetof(osdConfig_t, distance_alarm) },
     { "osd_esc_temp_alarm",         VAR_INT8   | MASTER_VALUE, .config.minmax = { INT8_MIN, INT8_MAX }, PG_OSD_CONFIG, offsetof(osdConfig_t, esc_temp_alarm) },
-    { "osd_esc_rpm_alarm",          VAR_INT16  | MASTER_VALUE, .config.minmax = { ESC_RPM_ALARM_OFF, INT16_MAX }, PG_OSD_CONFIG, offsetof(osdConfig_t, esc_rpm_alarm) },
+    { "osd_esc_rpm_alarm",          VAR_UINT32 | MASTER_VALUE, .config.u32Max = 1000000, PG_OSD_CONFIG, offsetof(osdConfig_t, esc_rpm_alarm) },
```

Comment:
why disable `ESC_RPM_ALARM_OFF`?
```
# set osd_esc_rpm_alarm = -1
###ERROR IN set: INVALID VALUE###
Allowed range: 0 - 1000000
```

Reply:
I changed it by 0 because erpmToRpm has to return an uint32_t value, so I had to change osd_esc_rpm_alarm to also be uint32_t to avoid unsigned to signed comparison error when compiling.

Reply:
Yes we might need another struct:
```
typedef struct cliMinMaxUnsigned32Config_s {
    const uint32_t min;
    const uint32_t max;
} cliMinMaxUnsigned32Config_t;
```

Reply:
I have been checking cliValueConfig_t union that is the place where we might add the new structure. This union is used by clivalue_t that is used to have an array of all cli settings. I think that this union is a tradeoff between functionality and flash usage. Adding the proposed struct to the union will take additional 4 bytes each setting. So I think we have several possibilities to avoid increasing flash usage sticking to the current struct.

- Use current "cliMinMaxConfig_t minmax" member for osd_esc_rpm_alarm. This alarm triggers when rpm goes below the setting so we have [-1 - 32767] range of values. This way we will be able to keep value -1 for the disable value, and 0 to 32767 to define a threshold, that I think is a good range for this alarm. This way we can keep backwards compatibility.
- Use current u32Max member, 0 for disable, and a range between 1 to 1000000 to define a threshold. This way we lose backwards compatibility, but we have a full range to define a threshold.

In case you prefer to increase ranges for settings I recommend to update cliMinMaxConfig_t datatype to use int32_t values instead int16_t ones. This way defining a new struct is avoided, and it will increase ranges for new settings when necessary. Also we can keep value -1 to to keep backwards compatibility for osd_esc_rpm_alarm setting.

Please, what do you prefer to do?

Reply:
Option 1. 

If you want option 2 while retaining backwards compatibility we need to rewrite `cliMinMaxConfig_t`.
But at what costs in terms of flash size? Like to have other opinions here before we decide on this.

Reply:
Option 1. Adding or upgrading the struct with two max min 32 bit members increases flash size in 2416 bytes. I think it is reasonable to keep it as it is now, keeping backwards compatibility for osd_esc_rpm_alarm

With STM32F7X2 using 16 bit members for max/min:
```
Linking STM32F7X2 
Memory region         Used Size  Region Size  %age Used
        ITCM_RAM:       15936 B        16 KB     97.27%
      ITCM_FLASH:          0 GB        16 KB      0.00%
ITCM_FLASH_CONFIG:          0 GB        16 KB      0.00%
     ITCM_FLASH1:          0 GB       480 KB      0.00%
      AXIM_FLASH:        2896 B        10 KB     28.28%
AXIM_FLASH_CUSTOM_DEFAULTS:           8 B         6 KB      0.13%
AXIM_FLASH_CONFIG:          0 GB        16 KB      0.00%
     AXIM_FLASH1:      479977 B       480 KB     97.65%
AXIM_FLASH_CUSTOM_DEFAULTS_EXTENDED:          0 GB         0 GB
        DTCM_RAM:       41280 B        64 KB     62.99%
           SRAM1:       76112 B       176 KB     42.23%
           SRAM2:          0 GB        16 KB      0.00%
       MEMORY_B1:          0 GB         0 GB
   text	   data	    bss	    dec	    hex	filename
 475841	   7040	 110356	 593237	  90d55	./obj/main/betaflight_STM32F7X2.elf
Creating HEX ./obj/betaflight_4.4.0_STM32F7X2.hex 
make[2]: se sale del directorio '/home/blue/proyectos/betaflight'
make[1]: se sale del directorio '/home/blue/proyectos/betaflight'
Building STM32F7X2 succeeded.
```
With STM32F7X2 using 32 bit members for max/min:
```
Linking STM32F7X2 
Memory region         Used Size  Region Size  %age Used
        ITCM_RAM:       15936 B        16 KB     97.27%
      ITCM_FLASH:          0 GB        16 KB      0.00%
ITCM_FLASH_CONFIG:          0 GB        16 KB      0.00%
     ITCM_FLASH1:          0 GB       480 KB      0.00%
      AXIM_FLASH:        2896 B        10 KB     28.28%
AXIM_FLASH_CUSTOM_DEFAULTS:           8 B         6 KB      0.13%
AXIM_FLASH_CONFIG:          0 GB        16 KB      0.00%
     AXIM_FLASH1:      482393 B       480 KB     98.14%
AXIM_FLASH_CUSTOM_DEFAULTS_EXTENDED:          0 GB         0 GB
        DTCM_RAM:       41280 B        64 KB     62.99%
           SRAM1:       76112 B       176 KB     42.23%
           SRAM2:          0 GB        16 KB      0.00%
       MEMORY_B1:          0 GB         0 GB
   text	   data	    bss	    dec	    hex	filename
 478257	   7040	 110356	 595653	  916c5	./obj/main/betaflight_STM32F7X2.elf
```